src/scripts/lsf_submit.sh

```diff
@@ -142,6 +142,31 @@ else
 fi
 # --- End of MPI directives
 
+# --- Begin of GPU directives
+
+if [ ! -z $bls_opt_gpunumber ] && [ $bls_opt_gpunumber -gt 0 ] ; then
```

Comment:
is there a default value for these options? (`$bls_opt_gpunumber` and  `$bls_opt_gpunumber`)

if not, this looks to me like it will blow chunks if these vars are empty/unset

Reply:
I pasted that code block into a bare bash script and it happily treated the conditional as false.

Reply:
To me it's one of those "i do not think that word means what you think it means" moments.

You are right, it works if `bls_opt_gpunumber` is unset, though only incidentally.

`[ ! -z $bls_opt_gpunumber ]` becomes `[ ! -z ]` which means, 'test that the length of the string "-z" is not non-zero', which is false; thus `[ $bls_opt_gpunumber -gt 0 ]` (which would blow chunks) is not evaluated at all due to the `&&` short circuit.

So, incidentally, the behavior is logically equivalent to what was intended.

So it's one of those "if you break it enough, it just might work" kind of things.

Anyway, i only give my approval in the "obfuscated code contest" sense.